src/extract.rs

```diff
@@ -0,0 +1,76 @@
+use std::ops::RangeBounds;
+
+use crate::TableOption;
+
+/// Returns a new [Table] that reflects a segment of the referenced [Table]
+///
+/// The segment is defined by [RangeBounds<usize>] for Rows and Columns
+///
+/// ```rust,no_run
+/// Extract::new(rows: .., columns: ..)
+/// ```
+/// 
+/// # Example
+///
+/// ```
+/// use tabled::{Table, Format, Row, Modify};
+///
+/// let data = vec![
+///     (0, "Grodno", true),
+///     (1, "Minsk", true),
+///     (2, "Hamburg", false),
+///     (3, "Brest", true),
+/// ];
+///
+/// let table = Table::new(&data)
+///                .with(Modify::new(Row(1..)).with(Format(|s| format!(": {} :", s))))
+///                .with(Extract::new(1..=2, 1..))
+///                .to_string();
+///
+/// assert_eq!(table, "+-------------+-----------+\n\
+///                    | : Grodno :  | : true :  |\n\
+///                    +-------------+-----------+\n\
+///                    |  : Minsk :  | : true :  |\n\
+///                    +-------------+-----------+\n");
+///
+/// assert_ne!(table, "+-------+-------------+-----------+\n\
+///                    |  i32  |    &str     |   bool    |\n\
+///                    +-------+-------------+-----------+\n\
+///                    | : 0 : | : Grodno :  | : true :  |\n\
+///                    +-------+-------------+-----------+\n\
+///                    | : 1 : |  : Minsk :  | : true :  |\n\
+///                    +-------+-------------+-----------+\n\
+///                    | : 2 : | : Hamburg : | : false : |\n\
+///                    +-------+-------------+-----------+\n\
+///                    | : 3 : |  : Brest :  | : true :  |\n\
+///                    +-------+-------------+-----------+\n");
+/// ```
+///
+pub struct Extract<R, C>
+where
+    R: RangeBounds<usize>,
+    C: RangeBounds<usize>,
```

Comment:
As I understand you can just remove this block?

Reply:
In my testing you are right. These can be removed. Will do